R/writeImage.R

```diff
@@ -329,4 +329,46 @@ decodeColNames <- function(x, strict = FALSE, fun = NULL, ...) {
   return(x)
 }
 
+.applyEnDeCoder.call <- function(x, fun, ...) {
+  return(.modify_lang(x, fun, ...))
+}
+
+.modify_lang <- function(x, f, ...) {
+  # adapted from pryr::modify_lang
+  # changed modify_lang to .modify_lang and added an explicit is.character check before doing f
 
+  # @param x a call object
+  # @param f a function to apply to each leaf
+
+  recurse <- function(y) {
+    # if (!is.null(names(y))) names(y) <- f2(names(y))
+    lapply(y, .modify_lang, f = f, ...)
+  }
+
+  if (is.atomic(x) || is.name(x)) {
+    # Leaf
+    # modified bit:
+    if (is.character(x))
+      f(x, ...)
+    else
+      x
```

Comment:
This could be generalized:

```r
.applyEnDeCoder(x, fun, ...)
```
which would also dispatch to other methods than character if needed. It's just that I've never ever encountered non-character objects in expressions, and the recursion would get a bit longer. For example, instead of this stack: 

`.modify_lang` -> `.modify_lang` -> `modify_lang` (3 levels before reaching a character) 

you would get this stack: 

`modify_lang.` -> `.applyEnDeCoder` -> `applyEnDeCoder.call` -> `modify_lang.` -> `.applyEnDeCoder` -> `applyEnDeCoder.call` -> `modify_lang.` -> `.applyEnDeCoder` -> `applyEnDeCoder.character`.

I opted not to do this but if other people feel strongly about this I'm happy to change this!

Reply:
Well, I don't think the longer recursion would be a problem but also: if this works then that is good enough.

Also, having non-character objects in there does sound a bit weird and I guess the errormsg will make it clear enough that that is happening.